experiments/download_ssl4eo.py

```diff
@@ -0,0 +1,738 @@
+""" Sample and download Sentinel-1/2 tiles with Google Earth Engine
+
+#### run the script:
+# resample new ids with grid overlap search
+python ssl4eo_s12_downloader.py \
+    --save_path ./data \
+    --num_workers 8 \
+    --cloud_pct 20 \
+    --log_freq 100 \
+    --indices_range 0 250000
+
+### resume from interruption
+python ssl4eo_s12_downloader.py \
+    --save_path ./data \
+    --num_workers 8 \
+    --cloud_pct 20 \
+    --log_freq 100 \
+    --resume ./data/checked_locations.csv \
+    --indices_range 0 250000
+
+"""
+
+import argparse
+import csv
+import json
+import math
+import os
+import time
+import warnings
+from collections import OrderedDict
+from datetime import date, datetime, timedelta
+from multiprocessing.dummy import Lock, Pool
+from typing import Any, Dict, List, Optional, Tuple
+
+import ee
+import numpy as np
+import rasterio
+import shapefile
+import urllib3
+from rasterio.transform import Affine
+from shapely.geometry import Point, shape
+from torchvision.datasets.utils import download_and_extract_archive
+from tqdm import tqdm
+
+warnings.simplefilter("ignore", UserWarning)
+
+ALL_BANDS_L2A = [
+    "B1",
+    "B2",
+    "B3",
+    "B4",
+    "B5",
+    "B6",
+    "B7",
+    "B8",
+    "B8A",
+    "B9",
+    "B11",
+    "B12",
+]
+ALL_BANDS_L1C = [
+    "B1",
+    "B2",
+    "B3",
+    "B4",
+    "B5",
+    "B6",
+    "B7",
+    "B8",
+    "B8A",
+    "B9",
+    "B10",
+    "B11",
+    "B12",
+]
+RGB_BANDS = ["B4", "B3", "B2"]
+ALL_BANDS_GRD = ["VV", "VH"]
+
+
+""" samplers to get locations of interest points"""
+
+
+class UniformSampler:
+    def sample_point(self) -> List[float]:
+        lon = np.random.uniform(-180, 180)
+        lat = np.random.uniform(-90, 90)
+        return [lon, lat]
+
+
+class GaussianSampler:
+    def __init__(
+        self,
+        interest_points: Optional[List[List[float]]] = None,
+        num_cities: int = 1000,
+        std: float = 20,
+    ) -> None:
+        if interest_points is None:
+            cities = self.get_world_cities()
+            self.interest_points = self.get_interest_points(cities, size=num_cities)
+        else:
+            self.interest_points = interest_points
+        self.std = std
+
+    def sample_point(self) -> List[float]:
+        rng = np.random.default_rng()
+        point = rng.choice(self.interest_points)
+        std = self.km2deg(self.std)
+        lon, lat = np.random.normal(loc=point, scale=[std, std])
+        return [lon, lat]
+
+    @staticmethod
+    def get_world_cities(
+        download_root: str = os.path.expanduser("./world_cities/"),
+    ) -> List[Dict[str, Any]]:
+        url = "https://simplemaps.com/static/data/world-cities/basic/simplemaps_worldcities_basicv1.71.zip"  # noqa
```

Comment:
```suggestion
        url = "https://simplemaps.com/static/data/world-cities/basic/simplemaps_worldcities_basicv1.71.zip"  # noqa: E501
```
This is more specific so it only ignores "line too long" errors and nothing else